### PR TITLE
Merge CPUBufferOpsDefault pipeline into CPUDefault pipeline.

### PR DESCRIPTION
**Pull request by hanhanW**
We no longer need CPUBufferOpsDefault pipeline because bufferization is
able to handle early bufferized kernels.

This needs to add bufferization::ToMemrefOp to deny list. This kind of
ops are created by bufferization, it's safe to exclude it if running the
same bufferization twice.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Dialect/LoweringConfig.td [hunk 1]
```diff
@@ -11,62 +11,57 @@ include "iree/compiler/Codegen/Dialect/IREECodegenDialect.td"
...
+    : I32EnumAttrCase<"CPUDoubleTilingPadExpert", 2>;
 def CPU_ConvTileAndDecomposeExpert
-    : I32EnumAttrCase<"CPUConvTileAndDecomposeExpert", 4>;
+    : I32EnumAttrCase<"CPUConvTileAndDecomposeExpert", 3>;
 def CPU_TileFuseAndVectorize
-    : I32EnumAttrCase<"CPUTileFuseAndVectorize", 5>;
+    : I32EnumAttrCase<"CPUTileFuseAndVectorize", 4>;
 def CPU_BufferOpsTileAndVectorize
-    : I32EnumAttrCase<"CPUBufferOpsTileAndVectorize", 6>;
+    : I32EnumAttrCase<"CPUBufferOpsTileAndVectorize", 5>;
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Codegen/Dialect/LoweringConfig.td:25]: I really don't like this kind of change... Do we have something similar to C++ enum in MLIR?

E.g.,

```
enum LoweringConfig : int32_t {
  CPUDefault = 0,
  CPUDoubleTilingExpert,
 ...
  SPIRVVectorizeWithWorkgroupMemory,
  None = 0xFF
};
```

**Review by MaheshRavishankar — APPROVED** [review]
Very Nice! Thanks!

**Comment by hanhanW** [comment]
@matthias-springer The suggestion about excluding `bufferization::ToMemrefOp` works well in early bufferized cases. It works well for release build, but not debug build. In debug build, it hits an assertion:

```
Assertion `!inplaceBufferized.contains(&operand) && "OpOperand was already decided to bufferize inplace"' failed.
```

How do we deal with this? Can we remove the assertion and just do nothing if the operation if the operand is not contained in the inPlaceBufferized object?

Here's the IR before bufferization:

```mlir
module {
  func.func @_concatenate_cst_dispatch_1() {
    %cst = arith.constant dense<0> : tensor<2x3xi32>
    %c0 = arith.constant 0 : index
    %c2 = arith.constant 2 : index
    %c3 = arith.constant 3 : index
    %0 = bufferization.to_memref %cst : memref<2x3xi32, affine_map<(d0, d1)[s0, s1, s2] -> (d0 * s1 + s0 + d1 * s2)>>
    %1 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) offset(%c0) alignment(64) : memref<2x5xi32>
    memref.assume_alignment %1, 64 : memref<2x5xi32>
    %2 = memref.subview %1[0, 2] [2, 3] [1, 1] : memref<2x5xi32> to memref<2x3xi32, affine_map<(d0, d1) -> (d0 * 5 + d1 + 2)>>
    %workgroup_id_x = hal.interface.workgroup.id[0] : index
    %workgroup_count_x = hal.interface.workgroup.count[0] : index
    %workgroup_id_y = hal.interface.workgroup.id[1] : index
    %workgroup_count_y = hal.interface.workgroup.count[1] : index
    scf.for %arg0 = %workgroup_id_y to %c2 step %workgroup_count_y {
      %3 = affine.apply affine_map<()[s0] -> (s0 * 3)>()[%workgroup_id_x]
      %4 = affine.apply affine_map<()[s0] -> (s0 * 3)>()[%workgroup_count_x]
      scf.for %arg1 = %3 to %c3 step %4 {
        %5 = memref.subview %0[%arg0, %arg1] [1, 3] [1, 1] : memref<2x3xi32, affine_map<(d0, d1)[s0, s1, s2] -> (d0 * s1 + s0 + d1 * s2)>> to memref<1x3xi32, affine_map<(d0, d1)[s0, s1, s2] -> (d0 * s1 + s0 + d1 * s2)>>
        %6 = memref.subview %2[%arg0, %arg1] [1, 3] [1, 1] : memref<2x3xi32, affine_map<(d0, d1) -> (d0 * 5 + d1 + 2)>> to memref<1x3xi32, affine_map<(d0, d1)[s0] -> (d0 * 5 + s0 + d1)>>
        linalg.generic {indexing_maps = [affine_map<(d0, d1) -> (d0, d1)>, affine_map<(d0, d1) -> (d0, d1)>], iterator_types = ["parallel", "parallel"]} ins(%5 : memref<1x3xi32, affine_map<(d0, d1)[s0, s1, s2] -> (d0 * s1 + s0 + d1 * s2)>>) outs(%6 : memref<1x3xi32, affine_map<(d0, d1)[s0] -> (d0 * 5 + s0 + d1)>>) attrs =  {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[1, 3]]>} {
        ^bb0(%arg2: i32, %arg3: i32):
          linalg.yield %arg2 : i32
        }
      }
    }
    return
  }
}
```

**Comment by matthias-springer** [comment]
Just discussed offline with Mahesh. So you tried excluding the `bufferization.to_memref` op (in addition to `arith.const`) and you're still seeing this assertion.

This should not be happening and I can actually not reproduce it locally. (Sent you a CL on internal chat.) So I'm curious what's going on here...

I also prepared a patch that allows to_memref ops when there no other bufferizable ops: https://reviews.llvm.org/D125718 Then it is no longer needed to exclude that op.

**Comment by hanhanW** [comment]
Had an offline discussion with @matthias-springer , we should use the same options in init_tensor elimination. `testAnalysisOnly` and `printConflicts` should always be false in the elimination.

**Pull request merged**